app/tests/spec/views/marketing_snippet.js

```diff
@@ -42,88 +56,127 @@ define(function (require, exports, module) {
       view = windowMock = null;
     });
 
-    describe('render', function () {
-      it('normally shows sign up marketing material to non-FxMobile sync users', function () {
-        windowMock.navigator.userAgent = 'Mozilla/5.0 (Windows NT x.y; rv:31.0) Gecko/20100101 Firefox/31.0';
+    testMarketingCampaign(Constants.MARKETING_ID_SPRING_2015);
 
-        createView();
+    function testMarketingCampaign (marketingId) {
+      function testLinkEl(view, expectedType) {
```

Comment:
nit: `testLinkEl` feels like a bad name. Can this be: `assertLinkHasExpectedType` or something?